bottles/frontend/widgets/library.py

```diff
@@ -74,6 +79,15 @@ def __init__(self, library, uuid, entry, *args, **kwargs):
 
         if entry.get('thumbnail'):
             path = ThumbnailManager.get_path(self.config, entry['thumbnail'])
+
+            if path is None:
+                # redownloading *should* never fail as it was successfully downloaded before
```

Comment:
do we assume that the connection is accessible?

Reply:
I did make this assumption... Will change it!